catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/TableRepository.java

```diff
@@ -463,14 +464,24 @@ private EntityReference getDatabase(UUID tableId) throws IOException {
     return daoCollection.databaseDAO().findEntityReferenceById(UUID.fromString(result.get(0)));
   }
 
-  private EntityReference getLocation(UUID tableId) throws IOException {
+  private EntityReference getLocation(Table table) throws IOException {
     // Find the location of the table
     List<String> result =
         daoCollection
             .relationshipDAO()
-            .findTo(tableId.toString(), Entity.TABLE, Relationship.HAS.ordinal(), Entity.LOCATION);
+            .findTo(
+                table.getId().toString(),
+                Entity.TABLE,
+                Relationship.HAS.ordinal(),
+                Entity.LOCATION,
+                toBoolean(Include.ALL));
     if (result.size() == 1) {
-      return daoCollection.locationDAO().findEntityReferenceById(UUID.fromString(result.get(0)));
+      Location location = daoCollection.locationDAO().findEntityById(UUID.fromString(result.get(0)));
+      if (!table.getDeleted() && location.getDeleted()) {
+        return null;
+      } else {
+        return daoCollection.locationDAO().getEntityReference(location);
+      }
```

Comment:
Removing a location from a table is hard deleting the relationship.
Deleting a location is soft deleting the relationship.
If both the table and the location are soft deleted, it makes sense to show the relationship.
#
![image](https://user-images.githubusercontent.com/32617/149064832-4f629157-5307-47d8-ba5b-fe21d24a3432.png)

Reply:
I'm closing for now. Just reopen it if needed.